openshift/e2e-tests-openshift.sh

```diff
@@ -47,173 +54,118 @@ function install_strimzi(){
 }
 
-function install_istio(){
-  header "Installing Istio"
+function install_servicemesh(){
+  header "Installing ServiceMesh"
 
-  # Install the Maistra Operator
-  oc new-project istio-operator
-  oc new-project istio-system
-  oc apply -n istio-operator -f https://raw.githubusercontent.com/Maistra/istio-operator/maistra-${MAISTRA_VERSION}/deploy/maistra-operator.yaml
+  # Install the ServiceMesh Operator
+  oc apply -f https://raw.githubusercontent.com/openshift/knative-serving/release-${SERVING_VERSION}/openshift/servicemesh/operator-install.yaml
 
-  # Wait until the Operator pod is up and running
-  wait_until_pods_running istio-operator || return 1
+  # Wait for the istio-operator pod to appear
+  timeout 900 '[[ $(oc get pods -n openshift-operators | grep -c istio-operator) -eq 0 ]]' || return 1
 
-  # Workaround for MAISTRA-670
-  oc delete validatingwebhookconfiguration istio-operator.servicemesh-resources.maistra.io
+  # Wait until the Operator pod is up and running
+  wait_until_pods_running openshift-operators || return 1
 
-  # Deploy Istio
+  # Deploy ServiceMesh
+  oc new-project $SERVICEMESH_NAMESPACE
+  oc apply -n $SERVICEMESH_NAMESPACE -f https://raw.githubusercontent.com/openshift/knative-serving/release-${SERVING_VERSION}/openshift/servicemesh/controlplane-install.yaml
   cat <<EOF | oc apply -f -
 apiVersion: maistra.io/v1
-kind: ServiceMeshControlPlane
-metadata:
-  name: minimal-multitenant-cni-install
-  namespace: istio-system
-spec:
-  istio:
-    global:
-      multitenant: true
-      proxy:
-        autoInject: disabled
-      omitSidecarInjectorConfigMap: true
-      disablePolicyChecks: false
-    istio_cni:
-      enabled: true
-    gateways:
-      istio-ingressgateway:
-        autoscaleEnabled: false
-        type: LoadBalancer
-      istio-egressgateway:
-        enabled: false
-      cluster-local-gateway:
-        autoscaleEnabled: false
-        enabled: true
-        labels:
-          app: cluster-local-gateway
-          istio: cluster-local-gateway
-        ports:
-          - name: status-port
-            port: 15020
-          - name: http2
-            port: 80
-            targetPort: 80
-          - name: https
-            port: 443
-    mixer:
-      enabled: false
-      policy:
-        enabled: false
-      telemetry:
-        enabled: false
-    pilot:
-      # disable autoscaling for use in smaller environments
-      autoscaleEnabled: false
-      sidecar: false
-    kiali:
-      enabled: false
-    tracing:
-      enabled: false
-    prometheus:
-      enabled: false
-    grafana:
-      enabled: false
-    sidecarInjectorWebhook:
-      enabled: false
----
-apiVersion: maistra.io/v1
 kind: ServiceMeshMemberRoll
 metadata:
   name: default
+  namespace: ${SERVICEMESH_NAMESPACE}
 spec:
   members:
-  - serving-tests
-  - serving-tests-alt
-  - knative-serving
+  - ${SERVING_NAMESPACE}
+  - ${EVENTING_NAMESPACE}
```

Comment:
@maschmid does this shitty config require all "tests" namespaces from eventing e2e ?

Reply:
yes, unfortunately... 

I've been using this:

```
readTestFiles() {
	for test in "./test/e2e"/*_test.go; do
		grep "func Test" $test | awk '{print $2}' | awk -F'(' '{print $1}' | sed "s/\([A-Z]\)/-\L\1/g" | sed "s/^-//"
		# expand function names with expected suffix, (this may generate test variants even for tests that don't have one, but it just means some extra namespace is created, so we don't care)
		for suffix in "-in-memory" "-crd-in-memory" "-crd-kafka" "-kafka"; do
			grep "func Test" $test | awk '{print $2}' | awk -F'(' '{print $1}' | sed "s/\([A-Z]\)/-\L\1/g" | sed "s/^-//" | awk -v suffix=$suffix '{print $1 suffix}'
		done
	done
}
```